### DISCUSSION TITLE
🤔: サンプルとしてFirebaseを使っている理由が知りたいです

### DISCUSSION THREAD

**Discussion by ghost** [ws-4020/mobile-app-crib-notes#111] [Q&A] [answered]
今後ＰＪでどれを使うべきか考えるときのヒントになりそうだと思い、質問したいのですが、
FCM, AWS Pinpoint, Azure Notification Hubsが紹介されていますが、サンプルとしてFirebaseが選定された理由が知りたいです。

※ コンテンツを読んでもらった後の質問を転記

**Comment by ghost** [comment, marked as answer]
Firebaseにしたのは、他のサービスを使う場合にもFirebaseは契約（無料）する必要があるためです。

他のサービスを利用する場合、コンテンツで説明している「Server→FCM→Device」という構造は「Server→[Some service]→FCM→Device」となります。

コンテンツとしては、プッシュ通知の基本的な仕組みを説明したかったのでFCMを題材としました。他のサービスとの機能的な比較によってFCMを題材としたわけではないので、プロジェクトで例えば「日時指定でのプッシュ通知」などFCMで提供されない機能が必要かどうかなどに応じて適切なサービスを検討する必要があります🙇‍♂️

**Comment by creature-water-valley** [comment]
回答済みのためクローズ
